retrieval/scrape.go

```diff
@@ -120,14 +120,24 @@ func newScrapePool(ctx context.Context, cfg *config.ScrapeConfig, app Appendable
 		// Any errors that could occur here should be caught during config validation.
 		log.Errorf("Error creating HTTP client for job %q: %s", cfg.JobName, err)
 	}
+
+	newLoop := func(
```

Comment:
It returns a `loop` while the inner function returns a `*scrapeLoop`. Thus we need this to have the right function signature.